app/models/flat.rb

```diff
@@ -1,4 +1,6 @@
 class Flat < ApplicationRecord
   belongs_to :user
 
+  validates :name, uniqueness: true
+  validates :rooms, :price, presence: true
```

Comment:
We should also validate the presence of user_id. A flat needs to have an owner.

Reply:
Thanks - updated this now